game/guitargraph.cc

```diff
@@ -160,13 +160,17 @@ void GuitarGraph::setupJoinMenuDifficulty() {
 	}
 	m_selectedDifficulty = ConfigItem(ol); // Create a ConfigItem from the option list
 	m_selectedDifficulty.select(cur); // Set the selection to current level
-	m_menu.add(MenuOption("", _("Select difficulty")).changer(m_selectedDifficulty)); // MenuOption that cycles the options
+	auto _selectDifficulty = std::make_unique<MenuOption>("", _("Select difficulty"));
```

Comment:
m_menu.add(std::make_unique<MenuOption>("", _("Select difficulty")).changer(m_selectedDifficulty));  ? why do you need to split that in 3 lines ?